extendr-api/src/lib.rs

```diff
@@ -6,59 +6,58 @@
 //!
 //! Anyone who knows the R library should be able to write R extensions.
 //! 
-//! See the [Robj](../robj/enum.Robj.html) struct for much of the content of this crate.
-//! [Robj](../robj/enum.Robj.html) provides a safe wrapper for the R object type.
+//! See [Robj] for much of the content of this crate.
+//! [Robj] provides a safe wrapper for the R object type.
 //! 
-//! This library is just being born, but goals are:
+//! Use attributes and macros to export to R.
+//! ```
+//! use extendr_api::*;
+//! // Export a function or impl to R.
+//! #[extendr]
+//! fn fred(a: i32) -> i32 {
+//!     a + 1
+//! }
 //!
-//! Implement common R functions such as c() and print()
+//! // define exports using extendr_module 
+//! extendr_module! {
+//!    mod mymodule;
+//!    fn fred;    
+//! }
+//! 
+//! ```
 //!
-//! Example:
+//! In R:
 //!
 //! ```ignore
-//! let v = c!(1, 2, 3);
-//! let l = list!(a=1, b=2);
-//! print!(v, l);
+//! result <- fred(1)
 //! ```
 //!
-//! Provide a wrapper for r objects.
+//! Common R functions such as c() and print() are available as macros:
 //!
-//! Example:
+//! ```
+//! use extendr_api::*;        // Put API in scope.
+//! extendr_engine::start_r(); // Start test environment.
+//! let v = c!(1., 2., 3.);
+//! let l = list!(a=1, b=2);
+//! assert_eq!(v, [1., 2., 3.].iter().collect_robj());
+//! ```
 //!
-//! ```ignore
+//! [Robj] is a wrapper for R objects.
+//! ```
+//! use extendr_api::*;        // Put API in scope.
+//! extendr_engine::start_r(); // Start test environment.
 //! let s = Robj::from("hello");
 //! let i = Robj::from(1);
 //! let r = Robj::from(1.0);
 //! ```
 //!
-//! Provide iterator support for creation and consumption of r vectors.
+//! Use iterators where possible instead of Vec and slices:
 //!
-//! Example:
-//!
-//! ```ignore
-//! let res = (1..=100).iter().collect::<Robj>();
-//! for x in res {
-//!     print!(x);
-//! }
-//! ```
-//!
-//! Provide a procedural macro to adapt Rust functions to R
-//!
-//! Example:
-//!
-//! ```ignore
-//! #[extendr]
-//! fn fred(a: i32) -> i32 {
-//!     a + 1
-//! }
 //! ```
-//!
-//! In R:
-//!
-//! ```ignore
-//!
-//! result <- .Call("fred", 1)
-//!
+//! use extendr_api::*;        // Put API in scope.
+//! extendr_engine::start_r(); // Start test environment.
+//! let res = (1..=100).collect_robj();
```

Comment:
Should this be `1..100`?

Reply:
The direct interpretation of `1:100` is `(1..=100)`, so I think this is appropriate here. But, probably we should also show `1..100` and explain it doesn't include `100` (This was one of the things I felt puzzling when I was a very beginner of Rust...).

Reply:
I see. Shows how little I know about Rust still.